### ISSUE TITLE
fails to redraw rows correctly following some resize events

### ISSUE THREAD

**Issue by y-ack** [12Me21/12term#1]
windows, wsl through vcxsrv
1. run
2. ls
3. snap to left side of screen
4. text before new prompt has disappeared
![text not redrawn correctly](https://user-images.githubusercontent.com/12588017/124524618-96dd4c00-ddb0-11eb-8548-31561ba9f1c4.png)
5. unsnap window
6. ls
7. increase window height
8. decrease window height
9. rows become cloned
![image](https://user-images.githubusercontent.com/12588017/124524692-dd32ab00-ddb0-11eb-8ec6-18a360769378.png)
10. press enter
11. decrease window height more
12. crash: corrupted double-linked list

log 1:
```
...
dirty rows: [####################] @ 0.81 ms [+0.81 ms] redraw
dirty rows: [...................#] @ 0.48 ms [+0.48 ms] redraw
dirty rows: [...................#] @ 0.23 ms [+0.23 ms] redraw
dirty rows: [###################] @ 1.49 ms [+1.49 ms] redraw
dirty rows: [##################] @ 1.58 ms [+1.58 ms] redraw
dirty rows: [#################] @ 1.96 ms [+1.96 ms] redraw
corrupted double-linked list
Aborted
```
log 2:
```
@ 24.52 ms [+24.52 ms] x stuff 1
@ 36.98 ms [+12.45 ms] parsed font pattern
@ 44.63 ms [+7.65 ms] loaded font 0
@ 52.79 ms [+8.16 ms] loaded font 2
@ 66.60 ms [+13.81 ms] loaded font 3
@ 82.57 ms [+15.96 ms] loaded font 1
@ 511.42 ms [+428.85 ms] created window
init
@ 516.83 ms [+5.41 ms] window mapped
@ 516.95 ms [+0.13 ms] first text
dirty rows: [##########] @ 0.33 ms [+0.33 ms] redraw
@ 10.34 ms [+10.01 ms] first draw
dirty rows: [##########] @ 0.32 ms [+0.32 ms] redraw
dirty rows: [##########] @ 0.29 ms [+0.29 ms] redraw
dirty rows: [.#........] @ 0.21 ms [+0.21 ms] redraw
dirty rows: [..........] @ 0.10 ms [+0.10 ms] redraw
dirty rows: [##########] @ 0.60 ms [+0.60 ms] redraw
dirty rows: [.........#] @ 0.31 ms [+0.31 ms] redraw
dirty rows: [#################################################################] @ 2.83 ms [+2.83 ms] redraw
dirty rows: [#################################################################] @ 3.14 ms [+3.14 ms] redraw
dirty rows: [#################################################################] @ 3.31 ms [+3.31 ms] redraw
dirty rows: [##########] @ 0.44 ms [+0.44 ms] redraw
dirty rows: [#################################################################] @ 2.39 ms [+2.39 ms] redraw
dirty rows: [##########] @ 1.02 ms [+1.02 ms] redraw
dirty rows: [##########] @ 0.92 ms [+0.92 ms] redraw
dirty rows: [##########] @ 0.28 ms [+0.28 ms] redraw
dirty rows: [.........#] @ 0.32 ms [+0.32 ms] redraw
dirty rows: [###########] @ 0.33 ms [+0.33 ms] redraw
dirty rows: [...........] @ 0.38 ms [+0.38 ms] redraw
dirty rows: [###########] @ 0.28 ms [+0.28 ms] redraw
dirty rows: [############] @ 0.44 ms [+0.44 ms] redraw
dirty rows: [............] @ 0.33 ms [+0.33 ms] redraw
dirty rows: [#############] @ 0.37 ms [+0.37 ms] redraw
dirty rows: [.............] @ 0.61 ms [+0.61 ms] redraw
dirty rows: [############] @ 0.48 ms [+0.48 ms] redraw
dirty rows: [###########] @ 1.10 ms [+1.10 ms] redraw
dirty rows: [##########] @ 0.46 ms [+0.46 ms] redraw
dirty rows: [##########] @ 0.16 ms [+0.16 ms] redraw
dirty rows: [##########] @ 0.30 ms [+0.30 ms] redraw
Segmentation fault```

**Comment by 12Me21** [comment]
oh github notifications are fucked so I didn't notice this..
i think it's fixed by #9d19e1e
edit: wow cool, referencing commits literally doesn't work

**Comment by y-ack** [comment]
fixed by 9d19e1e?
